### PR TITLE
Everywhere: Fix badly-formatted and unused includes

### PR DESCRIPTION
**Pull request by BenWiederhake**
This cleans up our includes:
- by fixing badly-formatted includes (e.g. `#include "AK/StdLibExtras.h"`)
- by removing completely unused includes.

This re-uses the least-controversial changes from #16764 and #15282, as well as a lot of new instances that have been introduced in the meantime.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/CanonicalIndex.h [hunk 1]
```diff
@@ -6,7 +6,6 @@
 
 #pragma once
 
-#include <AK/Concepts.h>
```
Comment (trflynn89) [inline Userland/Libraries/LibJS/Runtime/CanonicalIndex.h:9]: This is used.

File: Userland/Applications/Browser/History/HistoryModel.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 #pragma once
 
-#include "Applications/Browser/History.h"
+#include "../History.h"
```
Comment (trflynn89) [inline Userland/Applications/Browser/History/HistoryModel.h:9]: :eyes: IMO relative include paths like this are worse than just writing `<Applications/Browser/History.h>`, but I don't really have a stronger argument than "yuck".

File: Userland/Libraries/LibJS/Runtime/CanonicalIndex.h (hunk 1 above)
Comment (BenWiederhake) [reply Userland/Libraries/LibJS/Runtime/CanonicalIndex.h:9]: Oh right, `FloatingPoint`! You're right, I'll remove that. It compiles without any issue though.

File: Userland/Applications/Browser/History/HistoryModel.h (hunk 2 above)
Comment (BenWiederhake) [reply Userland/Applications/Browser/History/HistoryModel.h:9]: Yeah, that particular line is awful. I'll go with the system-include syntax like you suggested.

**Comment by BenWiederhake** [comment]
Changes since last push:
- Reverted change to LibJS/Runtime/CanonicalIndex.h (thanks @trflynn89!)
- Slightly nicer include in Browser/History/HistoryModel.h (thanks @trflynn89!)
- Rebased for convenience

**Pull request merged**